src/typechecker.erl

```diff
@@ -3721,6 +3721,10 @@ refine_ty(?type(nonempty_list, [Ty1]), ?type(nonempty_list, [Ty2]), Trace, Env)
         RefTy ->
             type(nonempty_list, [RefTy])
     end;
+refine_ty(?type(binary, [{integer, _, 0}, {integer, _, 8}]),
+          ?type(binary, [{integer, _, 0}, {integer, _, 0}]), _, _Env) ->
+    %% binary() \ <<>> => nonempty_binary()
+    type(binary, [{integer, 0, 8}, {integer, 0, 8}]);
```

Comment:
The `8` here can be turned into a variable N to make this a bit more generic. We just need a guard `when N > 0`.

Reply:
Done ✅